modules/apps/portal-search/portal-search-rest-impl/rest-openapi.yaml

```diff
@@ -1,64 +1,80 @@
 components:
     schemas:
-        Facet:
+        FacetConfiguration:
             properties:
                 aggregationName:
+                    description:
+                        The name of the aggregation.
                     type: string
                 attributes:
                     additionalProperties:
                         type: object
+                    description:
+                        Additional attributes needed by the facet type.
                     type: object
                 frequencyThreshold:
+                    description:
+                        Frequency threshold for showing the terms.
                     type: integer
                 maxTerms:
+                    description:
+                        Maximum number of terms to be shown.
                     type: integer
                 name:
+                    description:
+                        The name of the facet.
                     type: string
                 values:
+                    description:
+                        The values / selections to be filtered by.
                     items:
                         type: object
                     type: array
             type: object
         SearchRequestBody:
             properties:
-                facets:
-                    items:
-                        $ref: "#/components/schemas/Facet"
-                    type: array
-                searchContextAttributes:
+                attributes:
                     additionalProperties:
                         type: object
                     type: object
+                facetConfigurations:
+                    items:
+                        $ref: "#/components/schemas/FacetConfiguration"
+                    type: array
             type: object
-        SearchResponse:
+        SearchResult:
             properties:
-                aggregationResults:
+                actions:
```

Comment:
We're working on proving the actions automatically inside the DTOConverter. This means if you add actions here, you might get it also inside the embedded object too. I am not saying this is the wrong approach, because I understand you might want the actions even if you don't want the embedded object. Just pointing it out.

Reply:
Right, we don't always need the full object, but actions only here. The other thing was that I'm not using addAction() from superclass because that goes through the entity permissions checking which is not necessary, because the search already does that. Getting those alltogetger through the DTOConverter would be the best, but then we would again go through permissions one extra round?

Reply:
@4lejandrito would it be better to remove just to use the 'itemURL' instead of 'actions' on the SearchResult level? Semantically there shouldn't be any other actions available there. The full embedded entity could then get its actions through the DTOConverter (in the future)?

Reply:
Hey @peerkar:

> The other thing was that I'm not using addAction() from superclass because that goes through the entity permissions checking which is not necessary, because the search already does that.

One thing is the permissions to retrieve the specific element (VIEW) and another thing is the permissions needed to perform a specific action on it. Actions can contain things like `update`, `subscribe`... so we must check if the user has permissions to perform each of those actions. In the case of the `get`... yeah it would be redundant. 

> would it be better to remove actions and just to use the 'itemURL' instead of 'actions' (which I think was what you initially suggested) on the SearchResult level? Semantically there shouldn't be any other actions available there. The full embedded entity could then get its actions through the DTOConverter (in the future)?

I think this makes sense. Let's just have a `itemUrl` field and then if more actions are needed they will be available in the embedded object.